File: Java/Medium/WrongAnswer/minimum-garden-perimeter-to-collect-enough-apples_9.java
class Solution {
    public long minimumPerimeter(long neededApples) {
        
        long i = 0;
        while (true) {
            long apples = (i * 2 + 1) * (i * 2 + 1) * 3 / 2 + (i * 2 + 1) * 2;
            if (apples >= neededApples) {
                return (i * 2 + 1) * 4;
            }
            i++;
        }
    }
}

File: C++/Medium/WrongAnswer/minimum-garden-perimeter-to-collect-enough-apples_9.cpp
class Solution {
public:
    long long minimumPerimeter(long long neededApples) {
        
        long long n = sqrt(neededApples / 3.0) + 1;
        long long res = 0;
        for (int i = 1; i <= n; ++i) {
            long long cur = 1LL * (4 * i - 3) * (4 * i - 3);
            res += cur;
        }
        return res * 2;
    }
};